odl/operator/embedding.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2014, 2015 Jonas Adler
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators to cast between spaces."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.operator.operator import Operator
+
+__all__ = ('Embedding', 'EmbeddingFnInFn', 'EmbeddingPowerSpaceInFn',
+           'EmbeddingFnInPowerSpace')
+
+
+class Embedding(Operator):
+    """An operators to cast between spaces."""
+    def __init__(self, target, origin):
```

Comment:
And the "feel bad" vs "feel good" is the same as in math, where you have to think twice when your read `A o B` and need to map domains and ranges. I can see your point, reading as a "feed" from right to left is somewhat charming, but it interferes too much with the "math common sense".